dotnet/src/Connectors/Connectors.Memory.SqlServer/SqlServerServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,231 @@
+    /// <returns>The service collection.</returns>
+    public static IServiceCollection AddSqlServerCollection<TKey, TRecord>(this IServiceCollection services,
+        string collectionName,
+        string connectionString,
+        SqlServerCollectionOptions? options = null,
+        ServiceLifetime lifetime = ServiceLifetime.Singleton)
+        where TKey : notnull
+        where TRecord : class
+    {
+        Verify.NotNull(services);
+        Verify.NotNullOrWhiteSpace(collectionName);
+        Verify.NotNullOrWhiteSpace(connectionString);
+
+        services.Add(new ServiceDescriptor(typeof(SqlServerCollection<TKey, TRecord>),
+            sp => new SqlServerCollection<TKey, TRecord>(connectionString, collectionName, options), lifetime));
+
+        RegisterAbstractions<TKey, TRecord>(services, lifetime, serviceKey: null);
+
+        return services;
+    }
+
+    /// <summary>
+    /// Registers a keyed <see cref="SqlServerCollection{TKey, TRecord}"/> as <see cref="VectorStoreCollection{TKey, TRecord}"/>, with the specified connection string and service lifetime.
+    /// </summary>
+    /// <param name="services">The <see cref="IServiceCollection"/> to register the <see cref="VectorStoreCollection{TKey, TRecord}"/> on.</param>
+    /// <param name="serviceKey">The key with which to associate the collection.</param>
+    /// <param name="collectionName">The name of the collection.</param>
+    /// <param name="connectionString">The connection string.</param>
+    /// <param name="options">Options to further configure the collection.</param>
+    /// <param name="lifetime">The service lifetime for the store. Defaults to <see cref="ServiceLifetime.Singleton"/>.</param>
+    /// <returns>The service collection.</returns>
+    public static IServiceCollection AddKeyedSqlServerCollection<TKey, TRecord>(this IServiceCollection services,
+        object serviceKey,
+        string collectionName,
+        string connectionString,
+        SqlServerCollectionOptions? options = null,
+        ServiceLifetime lifetime = ServiceLifetime.Singleton)
+        where TKey : notnull
+        where TRecord : class
+    {
+        Verify.NotNull(services);
+        Verify.NotNull(serviceKey);
+        Verify.NotNullOrWhiteSpace(collectionName);
+        Verify.NotNullOrWhiteSpace(connectionString);
+
+        services.Add(new ServiceDescriptor(typeof(SqlServerCollection<TKey, TRecord>), serviceKey,
+            (sp, _) => new SqlServerCollection<TKey, TRecord>(connectionString, collectionName, options), lifetime));
+
+        RegisterAbstractions<TKey, TRecord>(services, lifetime, serviceKey);
+
+        return services;
+    }
+
+    private static void RegisterAbstractions<TKey, TRecord>(IServiceCollection services, ServiceLifetime lifetime, object? serviceKey)
+        where TKey : notnull
+        where TRecord : class
+    {
+        if (serviceKey is null)
```

Comment:
nit: the `serviceKey` parameter on TryAdd is nullable, so I think you can remove the condition and just pass through the parameter regardless of whether it's null or not.

Reply:
nice, I was not aware of that!